auxx/src/Command/Rollback.hs

```diff
@@ -34,41 +34,49 @@ import           Mode (MonadAuxxMode)
-                allTxs = concatMap extractTxs blunds
-            liftIO $ BSL.writeFile outFile (encode allTxs)
-            logInfo $ sformat ("Dumped "%int%" transactions to "%string)
-                      (length allTxs) (outFile)
-            rollbackBlocksUnsafe pm (BypassSecurityCheck True) (ShouldCallBListener True) blunds
-            logInfo $ sformat ("Rolled back "%int%" blocks") (length blunds)
-            printTipDifficulty
+rollbackAndDump pm pc numToRollback outFile =
+    withStateLock HighPriority ApplyBlockWithRollback $ \_ -> do
+        printTipDifficulty
+        blundsMaybeEmpty <- modifyBlunds
+            <$> DB.loadBlundsFromTipByDepth (fromIntegral numToRollback)
+        logInfo $ sformat ("Loaded " % int % " blunds")
+                          (length blundsMaybeEmpty)
+        case _Wrapped nonEmpty blundsMaybeEmpty of
+            Nothing     -> pass
+            Just blunds -> do
+                let extractTxs :: Blund -> [TxAux]
+                    extractTxs (Left _, _) = []
+                    extractTxs (Right mainBlock, _) =
+                        flattenTxPayload $ mainBlock ^. mainBlockTxPayload
+                let allTxs :: [TxAux]
+                    allTxs = concatMap extractTxs blunds
+                liftIO $ BSL.writeFile outFile (encode allTxs)
+                logInfo $ sformat
+                    ("Dumped " % int % " transactions to " % string)
+                    (length allTxs)
+                    (outFile)
+                rollbackBlocksUnsafe pm
+                                     pc
+                                     (BypassSecurityCheck True)
+                                     (ShouldCallBListener True)
+                                     blunds
+                logInfo $ sformat ("Rolled back " % int % " blocks")
+                                  (length blunds)
+                printTipDifficulty
   where
-    -- It's illegal to rollback 0-th genesis block.  We also may load
-    -- more blunds than necessary, because genesis blocks don't
-    -- contribute to depth counter.
+    -- It's illegal to rollback 0-th genesis block. We also may load more blunds
+    -- than necessary, because genesis blocks don't contribute to depth counter.
     modifyBlunds :: NewestFirst [] Blund -> NewestFirst [] Blund
     modifyBlunds =
         over _NewestFirst (genericTake numToRollback . skip0thGenesis)
     skip0thGenesis = filter (not . is0thGenesis)
     is0thGenesis :: Blund -> Bool
-    is0thGenesis (Left genBlock, _)
-        | genBlock ^. epochIndexL == 0 = True
-    is0thGenesis _ = False
+    is0thGenesis (Left genBlock, _) | genBlock ^. epochIndexL == 0 = True
+    is0thGenesis _                  = False
     printTipDifficulty = do
         tipDifficulty <- view difficultyL <$> DB.getTipHeader
-        logInfo $ sformat ("Our tip's difficulty is "%build) tipDifficulty
+        logInfo $ sformat ("Our tip's difficulty is " % build) tipDifficulty
```

Comment:
I personally don't like the lack of whitespace around the `%` operator, but Serokell's coding guidelines explicit state that the original version is the right one and the existing code base complies. If we decide to change this it should probably be something enforced by stylish-haskell.